azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+# The idea is to be inclusive of all source directories. (I.E. Commerce, Accounts, Eureka, Gateway, Inventory, Config, and Base(?))
+# And using the variable to set a condition that will set pipeline variables variables that will control the flow of the pipeline.
+
+trigger:
+
+  branches:
+    include: 
+    - experimental-pipeline
+
+pr:
+  branches:
+    include:
+    - main
+    - staging
+    - development
+    - feature_*
+    
+
+# Parameters and Variables to help with naming and calling variables, as well as providing a group of objects to loop over in later jobs.
+variables:
+    
+    org: '1054-daniel-felleman-revature-swag-shop'
+    projectKey: '1054-Daniel-Felleman-Revature-Swag-Shop_RevatureShopBackend'
+    projectName: 'RevatureShopBackend'
+
+parameters:
+- name: serviceConditions
+  type: object
+  default:
+  - accounts:
+    name: 'accounts'
+    requiresBase: true
+    jobName: 'AccountsBuild'
+  - commerce:
+    name: 'commerce'
+    requiresBase: true
+    jobName: 'CommerceBuild'
+  - config:
+    name: 'config'
+    jobName: 'ConfigBuild'
+  - eureka-server:
+    name: 'eureka-server'
+    jobName: 'EurekaBuild'
+  - inventory:
+    name: 'inventory'
+    requiresBase: true
+    jobName: 'InventoryBuild'
+  - Spring-Cloud-Gateway:
+    name: 'Spring-Cloud-Gateway'
+    jobName: 'SpringBuild'
+
+stages:
+- stage: Check
+  displayName: Check Service Changes, Test and Build
+  jobs:
+  - job: CheckChanges
+    pool: java-16
```

Comment:
there must be a way to install java 16 on the ms-hosted agents. this is ok but if we have self-hosted agents then we also want automation to set them up.

Reply:
Can look into this further, but the Azure Built in option is pretty rigid.

Reply:
you have sudo rights to install anything you want with apt, though

Reply:
Java 16 installs on agent. Found other errors along the way and fixed them.